### DISCUSSION TITLE
Can some one help me on how to call mockoon API from Python and get response

### DISCUSSION THREAD

**Discussion by lav20013** [mockoon/mockoon#951] [Help] [answered]
Can someone help me on how I can call mockoon API url from Python and get response Please?
Tutorial has setting up flask environment but doesn't provide any details on how mockoon API url can be called from Python

**Comment by 255kb** [comment, marked as answer]
This is not different from calling any other API. 
You can find a lot of tutorial on internet by searching for "Python API call". Some examples: 

https://www.educative.io/answers/how-to-make-api-calls-in-python
https://www.dataquest.io/blog/python-api-tutorial/

**Reply by lav20013** [reply]
I have tried calling from Python similar to other API as below
response = requests.get("http://localhost:3001/JobOrders",proxies=proxyDict ,verify=False)
print(response)
response i got is 400 not getting data , when I put the above url in browser it works , any thoughts what is wrong here?

**Reply by 255kb** [reply]
Could you share your mockoon data file/JSON? 
https://mockoon.com/docs/latest/mockoon-data-files/environment-clipboard-copy/

**Reply by lav20013** [reply]
{
    "uuid": "00f5ad39-ed7a-4ffb-884a-cc991b735355",
    "documentation": "",
    "method": "get",
    "endpoint": "JobOrders",
    "responses": [
        {
            "uuid": "f4d43005-0fce-4baa-a59b-75addcbbcf2d",
            "body": "{}",
            "latency": 0,
            "statusCode": 200,
            "label": "OpenJoborders",
            "headers": [],
            "bodyType": "FILE",
            "filePath": "C:\\PythonProject\\Mockoon\\JobOrders\\OpenJobOrder.json",
            "databucketID": "",
            "sendFileAsBody": true,
            "rules": [],
            "rulesOperator": "OR",
            "disableTemplating": false,
            "fallbackTo404": false,
            "default": true
        },
        {
            "uuid": "a01e134c-7e16-462d-b848-825a4c0090fb",
            "body": "{}",
            "latency": 0,
            "statusCode": 200,
            "label": "multiplejoborders",
            "headers": [],
            "bodyType": "FILE",
            "filePath": "C:\\PythonProject\\Mockoon\\JobOrders\\MultipleOpenJobOrders.json",
            "databucketID": "",
            "sendFileAsBody": true,
            "rules": [],
            "rulesOperator": "OR",
            "disableTemplating": false,
            "fallbackTo404": false,
            "default": false
        },
        {
            "uuid": "5db4ae46-5cb1-4ab5-9f60-9a42649af770",
            "body": "{}",
            "latency": 0,
            "statusCode": 200,
            "label": "cancelledjoborder",
            "headers": [],
            "bodyType": "FILE",
            "filePath": "C:\\PythonProject\\Mockoon\\JobOrders\\CancelledJobOrder.json",
            "databucketID": "",
            "sendFileAsBody": true,
            "rules": [],
            "rulesOperator": "OR",
            "disableTemplating": false,
            "fallbackTo404": false,
            "default": false
        }
    ],
    "enabled": true,
    "responseMode": null
}

**Reply by 255kb** [reply]
I don't see anything odd.
Could it be a proxy issue?
Is there any details worth the error code?
Could you also check Mockoon logs (top menu -> show logs)?

**Reply by lav20013** [reply]
Nothing in the logs our company does have proxy and i am passing in the get request as well , will try to ask team internally if they can help

**Reply by 255kb** [reply]
Ok.
One thing I could check, could you also share a JSON file you are serving?

**Reply by lav20013** [reply]
{"total": 1, "start": 0, "count": 1,
   "data": [{"id": 1161463,"publicDescription": "<p>test</p>\r\n", "title": "Automation1JPOpen", "dateAdded": 1634936601847,"startDate": 1633320000000,"dateEnd": 1646024400000,
        "dateLastModified": 1635537093930,"address": {"address1": "","address2": null,"city": "SmithField","countryCode": "US","countryID": 1,"countryName": "United States",
        "state": "Rhode Island","timezone": "America/Chicago","zip": "76262"},"customText11": "Full Time","customText7": "784","correlatedCustomText5": "Onsite",
        "isOpen": true,"status": "Open" }]
}

**Reply by lav20013** [reply]
Proxy caused the error had to set http and https proxies to  empty and pass proxy then it worked thank you for all the help.Mockoon was easy to setup and use it,looking forward for the ability to update crud operations option.

**Reply by 255kb** [reply]
Great! Glad you found the issue.
